cartridge/shop/managers.py

```diff
@@ -22,26 +22,20 @@ def from_request(self, request):
         cart.
         """
         cart_id = request.session.get("cart", None)
-        cart = None
-        if cart_id:
+        cart = self.current().filter(id=cart_id)
+        last_updated = now()
+        # Update timestamp and clear out old carts.
+        if cart_id and cart.update(last_updated=last_updated):
+            self.expired().delete()
+        else:
```

Comment:
In the old code, the session would only be modified if it previously contained a cart ID.

Can you see any problem with changing that ``else`` to an ``elif cart_id:`` to maintain that behaviour?